### PR TITLE
added yaml file support

### PR DESCRIPTION
**Pull request by arshad-k7**
Added yaml parsing support same as the toml parsing support 
issue https://github.com/vukani-dev/improvement-roll-service/issues/1

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -233,6 +235,17 @@ func parseCategory(filePath string, categorName string) SharedCategory {
 		toml.Unmarshal(byteValue, &sharedCategory)
 		return sharedCategory
 	}
+	if fileExtension == ".yaml"{
+		yamlFile, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			fmt.Println(err)
+			return SharedCategory{}
+		}
+		byteValue, _ := yaml.Marshal(yamlFile)
```
Comment (vukani-dev) [inline main.go]: I tried using a yaml file and the struct was empty.
I think it has to do with this.
Don't need to Marshal the file to bytes to Unmarshal yaml into the struct.
You can just Unmarshal the `yamlFile`.

File: main.go [hunk 2]
```diff
@@ -233,6 +235,17 @@ func parseCategory(filePath string, categorName string) SharedCategory {
...
+		var sharedCategory SharedCategory
+		yaml.Unmarshal(byteValue, &sharedCategory)
```
Comment (vukani-dev) [inline main.go]: Unmarshal the `yamlFile` instead of bytes

**Review by vukani-dev — CHANGES_REQUESTED** [review]
Ran into an issue while trying this. yaml files came through as empty on the API.

File: main.go (hunk 2 above)
Comment (arshad-k7) [reply main.go]: Resolved
and working now

File: main.go (hunk 1 above)
Comment (arshad-k7) [reply main.go]: resolved

**Comment by arshad-k7** [comment]
@vukani-dev can you please look once again?

**Comment by vukani-dev** [comment]
LGTM

**Pull request merged**